content/blog/2019/06/2019-06-25-Phase1-Updates-On-Working-Hours-Plugin

```diff
@@ -0,0 +1,113 @@
+
+* A standalone webapp which could be then integrated.
+* Slider for choosing a time range.
+* More fields when setting a excluded date.
+* Presets for choosing a excluded date.
+* A Jenkins styling UI.
+
+== How We Integrate React Into Jenkins
+
+A solution doc for integration could be found at 
+https://drive.google.com/open?id=1JLRCDg9JNBWR0Dfq8w3pTI9mrl6i9JU29pBoH6bO0J8
+
+At first, we found BlueOcean is a great example for using React in Jenkins, but yet it's not a choice for common development with plugins. So we need to find out another way to integrate.
+
+There are three steps to do the integration:
+
+* A mount point in your jelly file, usually it's a element with a unique `id`.
+* Write your React Application, but need to set the mount point to the id you set above.
+* Copy the output after you build the Project into the plugin's webapp dir.
+* Add your files using a script tag in your jelly file.
+[source, html]
+----
+<script type="text/javascript" src="${resURL}/plugin/working-hours/js/main.js"></script>
+---- 
+* Once we are using React, the traditional jelly request won't be available anymore, another way to process requests will be using stapler. You can define a process function like below.
+ [source, java]
+----
+public HttpResponse doDynamic(StaplerRequest request) {
+        if (config == null) {
+            config = ExtensionList.lookup(WorkingHoursPlugin.class).get(0);
+        }
+        String restOfPath = request.getRestOfPath();
+        String[] pathTokens = restOfPath.split("/");
+        List<String> params = new ArrayList<>();
+        switch (params.get(0)) {
+            case "list-excluded-dates":
+                return listExcludedDate(request);
+            case "set-excluded-dates":
+                return setExcludedDates(request);
+            case "list-time-ranges":
+                return listTimeRanges(request);
+            case "set-time-ranges":
+                return setTimeRanges(request);
+        }
+    }
+---- 
+
+== Run Our Application
+
+If you would like to take a look at our plugin, you can go to the repo
+link:https://github.com/jenkinsci/working-hours-plugin/tree/dev[working-hours-plugin]
+
+Just follow the README file, then you could run a copy of your working hours plugin.
+
+== Screenshots 
+
+Screenshot for Current Plugin
+
+image:/images/post-images/working-hours/working-hours-config.png[Screenshot for Current Plugin]
+
+Current
+
+Time Ranges
+
+image:/images/post-images/working-hours/working-hours-example-time-range.png[Time Ranges Example]
+
+Excludede Dates
+
+image:/images/post-images/working-hours/working-hours-exmaple-excluded-date.png[Excluded Dates Example]
+
+== Links May Help
+
+If you have any questions or advices, we are glad to hear from you.
+
+Some useful links is listed below:
+
+* link:https://github.com/jenkinsci/working-hours-plugin/tree/dev[Develop Repo]
```

Comment:
Is it needed at all? AFAICT it should be in pull requests

Reply:
This is the dev branch of the main repo, and the new codes will be merged into this branch via pull requests.

Reply:
ack, thanks